File: Questão2.py
class Carro:
    """Classe que representa um carro."""

    def __init__(self, marca: str, modelo: str, ano: int) -> None:
        self.marca = marca
        self.modelo = modelo
        self.ano = ano
        self.velocidade = 0

    def acelerar(self, incremento: int) -> None:
        self.velocidade += incremento

    def frear(self, decremento: int) -> None:
        self.velocidade = max(0, self.velocidade - decremento)

    def get_velocidade(self) -> int:
        return self.velocidade


carro = Carro("Toyota", "Corolla", 2020)
carro.acelerar(50)
print(f"Velocidade: {carro.get_velocidade()} km/h")
carro.frear(20)
print(f"Velocidade: {carro.get_velocidade()} km/h")

File: Questão2.java
class Carro {
    private String marca;
    private String modelo;
    private int ano;
    private int velocidade;

    public Carro(String marca, String modelo, int ano) {
        this.marca = marca;
        this.modelo = modelo;
        this.ano = ano;
        this.velocidade = 0;
    }

    public void acelerar(int incremento) {
        velocidade += incremento;
    }

    public void frear(int decremento) {
        velocidade = Math.max(0, velocidade - decremento);
    }

    public int getVelocidade() {
        return velocidade;
    }

    public static void main(String[] args) {
        Carro carro = new Carro("Toyota", "Corolla", 2020);
        carro.acelerar(50);
        System.out.println("Velocidade: " + carro.getVelocidade() + " km/h");
        carro.frear(20);
        System.out.println("Velocidade: " + carro.getVelocidade() + " km/h");
    }
}
